app/modules/individuals/models.py

```diff
@@ -383,6 +383,17 @@ def remove_names_for_value(self, value):
             name.delete()
         return num
 
+    def add_autogenerated_species_name(self, user):
+        from app.modules.autogenerated_names.models import AutogeneratedName
+
+        tx_guid = self.get_taxonomy_guid()
+        if not tx_guid:
+            return
+        autogen_names = AutogeneratedName.query.filter_by(reference_guid=tx_guid).all()
+        if autogen_names and len(autogen_names) == 1:
+            new_name = autogen_names[0].get_next()
+            self.add_name('autogen-species', new_name, user)
```

Comment:
not a blocking comment, but i wonder if this string used for the context on the actual name added should be derived from the `AutogeneratedNameType` values, like `auto_species`; just so we have a way to connect between the two without needing a new set of values for the name-context.  (tho i guess this value could be algorithmically generated from the AutogeneratedNameType, like `autogen-` + text after the `_`.)